### PR TITLE
Improve Galata

### PR DESCRIPTION
**Pull request by fcollonval**
## References

Extract improvement of Galata from https://github.com/jupyterlab/jupyterlab/pull/13240

## Code changes

The major change in Galata is the switch from runtime injection of the helper to a regular JupyterLab extension. This allows to get the helpers in the same namespace as JupyterLab - hence it gives us access to some non-token based elements like `Dialog`s.

Building on that, this introduces the ability to register listener to dialog and notifications events to make tests more robust; e.g. sometimes a test can be flaky because randomly some dialogs get in the way; with the new listener you could dismissed any random dialogs that may appear.

Adds new fixture `mockUser` to mock the user provided by the identity endpoint.

It renames `window.galataip` to `window.galata` as the later _ip_ is not easy to comprehend (it stands for in-page).

As the in-page helpers are now part of an extension, the server setup needs to evolve to add that extra extension. To ease future evolution of the server setup, a new helper `jupyterlab.galata.configure_jupyter_server` is provided.

## User-facing changes

None

## Backwards-incompatible changes

Remove `JupyterLab.notificationManager` trick used up to now to get access to notifications.
`window.galataip` is deprecated in favor of `window.galata`

### CODE REVIEW

File: galata/README.md [hunk 1]
```diff
@@ -81,6 +76,8 @@ Then start the server with:
 jupyter lab --config jupyter_server_test_config.py
 ```
 
+> If you need to customize the set up for galata, you can look at the `configure_jupyter_server <https://github.com/jupyterlab/jupyterlab/tree/master/jupyterlab/galata/__init__.py>`\_ definition.
```
Comment (fcollonval) [inline galata/README.md:79]:
```suggestion
> If you need to customize the set up for galata, you can look at the [`configure_jupyter_server`](https://github.com/jupyterlab/jupyterlab/tree/master/jupyterlab/galata/__init__.py) definition.
```

**Comment by fcollonval** [comment]
`check-links` is expected to fail due to a new link to a to-be-added file.

**Comment by jtpio** [comment]
Thanks this looks good :+1: 

The lint CI failure looks relevant though:

image

**Comment by fcollonval** [comment]
Thanks for the review @jtpio I fix the linter errors.

**Comment by jtpio** [comment]
Restarting the check release check, not sure it's relevant: 

image

**Comment by jtpio** [comment]
Looks like the check is now passing:

image


Maybe we'll want to get https://github.com/jupyterlab/jupyterlab/pull/13931 in first, to make sure the UI tests checks pass and this PR doesn't introduce regressions.

**Comment by fcollonval** [comment]
Merging to move forward - I keep working on #13931

**Pull request merged**